src/lib.rs

```diff
@@ -69,6 +72,14 @@
 //!     fn document_highlight(&self, _: TextDocumentPositionParams) -> Self::HighlightFuture {
 //!         Box::new(future::ok(None))
 //!     }
+//!
+//!     fn signature_help(&self,params: TextDocumentPositionParams) -> Self::SignatureHelpFuture {
+//!         Box::new(future::ok(None))
+//!     }
+//!
+//!     fn goto_implementation(&self,params: TextDocumentPositionParams) -> Self::GotoImplementationFuture {
```

Comment:
This method should be moved to immediately below `Backend::goto_type_definition()`.